deeplake/core/vectorstore/vector_search/utils.py

```diff
@@ -99,11 +100,26 @@ def parse_search_args(**kwargs):
         raise ValueError(
             f"Either an `embedding`, `embedding_function`, `filter`, or `query` must be specified."
         )
+
+    if kwargs["embedding"] is not None and kwargs["embedding_function"]:
+        always_warn(
+            "Both embedding data and embedding function were specified."
+            " Already computed `embedding` will be used."
+        )
     if kwargs["embedding_data"] is None and kwargs["embedding_function"] is not None:
         raise ValueError(
             f"When an `embedding_function` is specified, `embedding_data` must also be specified."
         )
 
+    if (
```

Comment:
is this logic correct? The error does not seem to match the check.

Reply:
yup, error message should say a vice versa